server/rest/geom.py

```diff
@@ -36,16 +39,26 @@ def __init__(self):
         self.route('DELETE', (':geomId',), self.deleteGeom)
         self.route('GET', ('export', ':itemId',), self.exportKPF)
 
+    # The girder default serialization takes twice the time than this custom serializer. Since geom could be relatively big, it worth to use this custom serializer. 
+    class JSONEncoder(json.JSONEncoder):
+        def default(self, obj):
+            if isinstance(obj, ObjectId):
+                return str(obj)
+            return json.JSONEncoder.default(self, obj)
+
     @autoDescribeRoute(
         Description('')
         .modelParam('itemId', model=Item, level=AccessType.READ)
         .errorResponse()
         .errorResponse('Read access was denied on the item.', 403)
     )
     @access.user
+    @rawResponse
     def getGeomsOfItem(self, item, params):
+        setResponseHeader('Content-Type', 'application/json')
         cursor = Geom().findByItem(item)
-        return list(cursor)
+        jsonString = self.JSONEncoder().encode(list(cursor))
```

Comment:
BTW, is there a reason to not just return this directly?

Reply:
I think this one is ok either way, I don't see anything wrong with assigning a value then return

Reply:
It's not *wrong*, per se, it just feels gratuitous... hence why the original comment is phrased as a question.